pkg/patch/internal/internalfakes/fake_mail_sender.go

```diff
@@ -0,0 +1,276 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by counterfeiter. DO NOT EDIT.
+package internalfakes
+
+import (
+	"sync"
+
+	"k8s.io/release/pkg/patch"
+)
+
+type FakeMailSender struct {
+	SendStub        func(string, string) error
+	sendMutex       sync.RWMutex
+	sendArgsForCall []struct {
+		arg1 string
+		arg2 string
+	}
+	sendReturns struct {
+		result1 error
+	}
+	sendReturnsOnCall map[int]struct {
+		result1 error
+	}
+	SetRecipientsStub        func(...string) error
+	setRecipientsMutex       sync.RWMutex
+	setRecipientsArgsForCall []struct {
+		arg1 []string
+	}
+	setRecipientsReturns struct {
+		result1 error
+	}
+	setRecipientsReturnsOnCall map[int]struct {
+		result1 error
+	}
+	SetSenderStub        func(string, string) error
+	setSenderMutex       sync.RWMutex
+	setSenderArgsForCall []struct {
+		arg1 string
+		arg2 string
+	}
+	setSenderReturns struct {
+		result1 error
+	}
+	setSenderReturnsOnCall map[int]struct {
+		result1 error
+	}
+	invocations      map[string][][]interface{}
+	invocationsMutex sync.RWMutex
```

Comment:
alternatively this could have been organized like so:
```
type someType1 struct {
  sync.RWMutex
  ...
}

type someType2 struct {
  sync.RWMutex
  ...
}

type formatter struct {
  someInstance1 someType1
  someInstance2 someType2
  ...
}
```

Reply:
same as https://github.com/kubernetes/release/pull/1040#discussion_r375186428